src/inc/callback.h

```diff
@@ -0,0 +1,14 @@
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
Can you please put the prototype into utilcode.h instead of creating a separate header? The utilcode.h is meant to expose functions implemented in utilcode lib and the function resides in there.